benchexec/tools/frama-c-sv.py

```diff
@@ -0,0 +1,43 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.util as util
+import benchexec.tools.template
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for Frama-C with a wrapper for usage in the SVCOMP.
+    URL: https://gitlab.com/sosy-lab/software/frama-c-sv
+    """
+
+    REQUIRED_PATHS = ["."]
+    _TOOL_NAME = "frama-c-sv"
+    _SCRIPT_NAME = _TOOL_NAME + ".py"
+
+    def executable(self, _):
+        return util.find_executable(_SCRIPT_NAME)
+
+    def program_files(self, executable):
+        return self._program_files_from_executable(
+            executable, self.REQUIRED_PATHS, parent_dir=True
+        )
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def name(self):
+        return _TOOL_NAME
+
+    def cmdline(self, executable, options, task, rlimits):
+        cmd = [_SCRIPT_NAME, "--program"] + list(task.input_files_or_identifier)
```

Comment:
You need to use `executable` here, this is why it is passed. Otherwise it will break for example if `_SCRIPT_NAME` is not in the current directory.

Does your tool support tasks with no or more than one input file? If the answer is no to a part of this question, please consider using `task.single_input_file` or `task.input_files` to produce nice error message for users.

Reply:
I had it as `BaseTool` module before and just rewrote it on the shortest possible path to fit the `BaseTool2` interface.

I probably just copied this from `cpachecker.py`.

Sure I will just use `task.input_files` then. Does CPAchecker support having no input file?

Reply:
Yes, it is used for example here: https://gitlab.com/sosy-lab/software/cpachecker/-/blob/trunk/test/test-sets/integration-value-java.xml